src/passes/RemoveUnusedBrs.cpp

```diff
@@ -906,6 +906,12 @@ struct RemoveUnusedBrs : public WalkerPass<PostWalker<RemoveUnusedBrs>> {
         // further optimizations after this, and those optimizations might even
         // benefit from this improvement.
         auto glb = Type::getGreatestLowerBound(curr->castType, refType);
+        if (!getModule()->features.hasCustomDescriptors() && glb.isExact() &&
+            !curr->castType.isExact()) {
+          // When custom descriptors is not enabled, nontrivial exact casts are
+          // not allowed.
+          glb = glb.with(Inexact);
```

Comment:
Perhaps we could have a utility `getMaybeWithoutExact(type, module)` which would check if the module's features support exactness and returns the type without, if not?

Reply:
Sounds good. Can I add that in a follow-up?

Reply:
Sure.